Dockerfile

```diff
@@ -49,31 +54,33 @@ ENV JULIA_PKG_USE_CLI_GIT="true"
 # path used during package precompilation matches the final depot path used in the image.
 # If a source file no longer resides at the expected location the `.ji` is deemed stale and
 # will be recreated.
-RUN ln -s /tmp/julia-cache ~/.julia
+ARG JULIA_DEPOT_CACHE=/mnt/julia-cache
+RUN ln -s ${JULIA_DEPOT_CACHE} ~/.julia
 
 # Install Julia package registries
-RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache,uid=1000,gid=100 \
-    mkdir -p /tmp/julia-cache && \
+RUN --mount=type=cache,sharing=locked,target=${JULIA_DEPOT_CACHE},uid=${UID},gid=${GID} \
+    mkdir -p ${JULIA_DEPOT_CACHE} && \
     julia -e 'using Pkg; Pkg.Registry.add("General")'
 
 # Instantiate the Julia project environment
-ENV JULIA_PROJECT /Ray.jl
-COPY --chown=ray *Project.toml *Manifest.toml ${JULIA_PROJECT}/
-
-# Generate a fake ray_julia_jll package just for instantiation
-RUN mkdir -p ${JULIA_PROJECT}/ray_julia_jll/src && touch ${JULIA_PROJECT}/ray_julia_jll/src/ray_julia_jll.jl
-
-# Note: The `timing` flag requires Julia 1.9
-RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache,uid=1000,gid=100 \
-    julia -e 'using Pkg; Pkg.Registry.update(); Pkg.instantiate(); Pkg.build(); Pkg.precompile(strict=true, timing=true)'
+ARG RAY_JL_PROJECT=${HOME}/.julia/dev/Ray
+COPY --chown=ray *Project.toml *Manifest.toml /tmp/Ray.jl/
+RUN --mount=type=cache,sharing=locked,target=${JULIA_DEPOT_CACHE},uid=${UID},gid=${GID} \
+    # Move project content into temporary depot
+    rm -rf ${RAY_JL_PROJECT} && \
+    mv /tmp/Ray.jl ${RAY_JL_PROJECT} && \
+    # Generate a fake ray_julia_jll package just for instantiation
+    mkdir -p ${RAY_JL_PROJECT}/ray_julia_jll/src && touch ${RAY_JL_PROJECT}/ray_julia_jll/src/ray_julia_jll.jl && \
+    # Note: The `timing` flag requires Julia 1.9
+    julia --project=${RAY_JL_PROJECT} -e 'using Pkg; Pkg.Registry.update(); Pkg.instantiate(); Pkg.build(); Pkg.precompile(strict=true, timing=true)'
```

Comment:
we no longer are building from source by default right?  Pkg.build shoudl be a no-op then, or do we intend to still actually build the JLLs?

Reply:
ah right at this stage, we're not doing any kind of real build because we've created a fake (empty) jll subdir.  however, when we swithc to making it a submodule, will this still work as intended?

Reply:
So `Pkg.build` is a no-op currently but it's best to include it in case a package depended on by Ray.jl requires it. 

> ah right at this stage, we're not doing any kind of real build because we've created a fake (empty) jll subdir. however, when we swithc to making it a submodule, will this still work as intended?

I'll need to do a couple of changes in the Dockerfile to accommodate the submodule refactor. I'll just be able to delete the creation of the fake JLL in this case